### PR TITLE
Support for bitcoin 0.17.x

### PR DESCRIPTION
**Pull request by gustavonalle**

### CODE REVIEW

**Comment by gustavonalle** [comment]
Tests are fine for Linux and MacOs, haven't tested on Windows

**Comment by gustavonalle** [comment]
related bitcoin-core issue:  https://github.com/bitcoin/bitcoin/issues/14082

File: eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala [hunk 1]
```diff
@@ -108,7 +108,8 @@ class Setup(datadir: File,
       } yield (progress, chainHash, bitcoinVersion, unspentAddresses)
       // blocking sanity checks
       val (progress, chainHash, bitcoinVersion, unspentAddresses) = Await.result(future, 30 seconds)
-      assert(bitcoinVersion.startsWith("16"), "Eclair requires Bitcoin Core 0.16.0 or higher")
+      val (versionInt, versionStr) = MinimumVersion
+      assert(bitcoinVersion > versionInt, s"Eclair requires Bitcoin Core $versionStr or higher")
```
Comment (conscott) [inline eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala]: Needs to be '>=' instead of '>' ?

Comment (gustavonalle) [reply eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala]: Fixed

**Comment by gustavonalle** [comment]
0.17.rc3 is out. I will update the PR and remove the TODOs

**Comment by gustavonalle** [comment]
PR updated:
* Removed the //TODOs
* It runs by default against bitcoin core ```0.16.0``` . This is dictated by the pom's property ```bitcoin-core.version```
* A new maven profile called ```0.17``` can be used to run against 0.17.x. In this PR, the profile is pointing to ```0.17.0rc3``` (the latest RC)
* Tested on windows, macOS and Linux the following executions, with all tests passing:
  * ```mvn clean install``` and ```mvn clean install -P0.17```

**Comment by gustavonalle** [comment]
Some versions of bitcoin-core disappeared  :) , I've moved to 0.17rc4 and 0.16.3

**Comment by gustavonalle** [comment]
Updated again, now pointing to the just released Bitcoin Core 0.17.0. 

[My PR to bitcoin-core](https://github.com/bitcoin/bitcoin/pull/14310) to fix the error reporting of ```signrawtransactionwithwallet``` for 0.17.x was merged, but for some reason it was not tagged for the release. That means the ```"unlock failed funding txes"``` test still needs to be adapted to work on both versions, but this is not a big deal.

**Pull request closed without merging**